compiler/ETA/CodeGen/Expr.hs

```diff
@@ -342,8 +342,13 @@ cgAlgAltRhss binder alts = do
                     | (DataAlt con, code) <- taggedBranches ]
   return (maybeDefault, branches)
 
-cgTick :: Tickish Id -> CodeGen ()
+cgTick :: Tickish a -> CodeGen ()
 cgTick (SourceNote srcSpan _) = do
-  let srcLoc = realSrcSpanStart srcSpan
-  addLineNumber $ srcLocLine srcLoc
+  mbCgSrcFile <- getSourceFileName
+  case mbCgSrcFile of
+    Just cgSrcFile -> do
+      let srcLoc  = realSrcSpanStart srcSpan
+          srcFile = fastStringText $ srcLocFile srcLoc
+      when (srcFile == cgSrcFile) $ addLineNumber $ srcLocLine srcLoc
```

Comment:
Will this condition hold for hierarchical modules?

From how you've defined it, for `GHC/List.hs`, `mgCgSrcFile` will be `Just "List.hs"` and `srcLocFile` will show up as `GHC/List.hs`. I think you'll need to store the *full* source file name as well in the CgState and do comparison against that.

Reply:
Ouch, you are right, i changed the way to get the source file just for this reason.
I'll fix asap